scripts/ado-terraform-nagger.py

```diff
@@ -272,155 +399,333 @@ def extract_version(text, regex):
-            log_message(
-                None,
-                None,
-                "warning",
-                f"Detected outdated terraform version: {terraform_version}",
-            )
-
-        # Handle terraform versions
-        terraform_version_checker(terraform_version)
-
-        # Handle providers
-        terraform_providers = result["provider_selections"]
-
-        for provider in terraform_providers:
-            if provider not in config["providers"]:
-                log_message(
+        # Retrieve the environment variables
+        system_default_working_directory = os.getenv('SYSTEM_DEFAULT_WORKING_DIRECTORY')
+        build_repo_suffix = os.getenv('BUILD_REPO_SUFFIX')
+
+        # Construct the working directory path
+        base_directory = os.getenv('BASE_DIRECTORY')
+        working_directory, components_list = create_working_dir_list(base_directory, system_default_working_directory, build_repo_suffix)
+        
+        output_warning = {
+            'terraform_version': {
+                'components': [],
+                'error_message': ''
+            },
+            'terraform_provider': {
+                'provider': {},
+                'error_message': ''
+            }
+        }
+
+        # for loop over dir componenets
+        for component in components_list:
+            print(f'COMPONENT: {component}')
+            full_path = f'{working_directory}{component}'
+            print(f'FULL PATH: {full_path}')
+
+            # Load deprecation map
+            config = load_file(args.filepath)
+
+            # Try to run `tfswitch' and 'terraform version --json` which is present in tf versions >= 0.13.0
+            command = ["tfswitch", "-b", terraform_binary_path]
+            run_command(command, full_path)
+
+            # try and do the terraform init 
+            command = ["terraform", "init", "-backend=false"]
+            output = run_command(command, full_path)
+            # check if tf has successfully init 
+            if not 'Terraform has been successfully initialized!' in output:
+                global errors_detected
+                relative_test_path = os.path.relpath(full_path, '../../../../../azp/_work/1/s')
+                logger.error(f'##vso[task.logissue type=error;]Repo structure invalid please see docs for further information: {relative_test_path}')
+                message = (f'Terraform initialized in an empty directory, Repo structure invalid please see docs for further information:\n{relative_test_path}')
+                errors_detected = True
+                log_message_slack(
                     slack_user_id,
                     slack_webhook_url,
-                    "warning",
-                    f"Provider {provider} is missing from version config. "
-                    "Please add it to the config in this file in order to "
-                    "compare it's versions.",
+                    message
                 )
+                raise SystemExit(1)
```

Comment:
Do we want to break the build if terraform init for version checker doesn't work? I think adding the warning /sending a message is enough

Reply:
Might also need to reconsider the `errors_detected = True` above this as well, as looks like that would raise a sys exit as well at the end of the script